### PR TITLE
Refactor InlineEditInput to functional component

### LINKED ISSUE #1519: Refactor `InlineEditInput` to be a functional component
and use hooks which will make it a bit easier to follow and implement elsewhere.

### PR DESCRIPTION
**Pull request by ManzoorWaniEE**
Fixes #1519 

## Checklist

* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [x] User input is adequately validated and sanitized
* [x] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.
* [] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/src/components/form/inputs/inline-edit-input/index.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { ENTER, ESCAPE, SPACE } from '@wordpress/keycodes';
+import { usePrevious } from '@eventespresso/hooks';
+
+/**
+ * Internal imports
+ */
+import './style.css';
+import Edit from './edit';
+import Display from './display';
```
Comment (tn3rb) [inline assets/src/components/form/inputs/inline-edit-input/index.js]: nice

File: assets/src/components/form/inputs/inline-edit-input/index.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+	useEffect( () => {
+		setOrigValue( inputValue );
+	}, [] );
+
+	const externalUpdate = async () => {
+		if (
+			! saving &&
+			prevState &&
+			'undefined' !== typeof prevState.value &&
```
Comment (tn3rb) [inline assets/src/components/form/inputs/inline-edit-input/index.js]: no Yoda conditionals please  :p

File: assets/src/components/form/inputs/inline-edit-input/index.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+			onKeyDown={ keySelect }
+			{ ...{
+				value,
+				valueType,
+				valueFormatter,
+				formatterSettings,
+				noticeStyle,
+			} }
+		/>;
+};
```
Comment (tn3rb) [inline assets/src/components/form/inputs/inline-edit-input/index.js]: plz re-add the prop-types that were part of the original class

File: assets/src/hooks/index.js [hunk 4]
```diff
@@ -34,3 +34,4 @@ export { default as useTicketsForEventDates }
 export { default as useTrashDateEntity } from './use-trash-date-entity';
 export { default as useTrashPriceModifier } from './use-trash-price-modifier';
 export { default as useTrashTicket } from './use-trash-ticket';
+export { default as usePrevious } from './use-previous';
```
Comment (tn3rb) [inline assets/src/hooks/index.js]: awesome thnx for putting this in here and not local to the inline edit input 👍

Comment (ManzoorWaniEE) [reply assets/src/hooks/index.js]: Yes, that way we can use it anywhere we want

File: assets/src/components/form/inputs/inline-edit-input/index.js (hunk 3 above)
Comment (ManzoorWaniEE) [reply assets/src/components/form/inputs/inline-edit-input/index.js]: Done

File: assets/src/components/form/inputs/inline-edit-input/index.js (hunk 2 above)
Comment (ManzoorWaniEE) [reply assets/src/components/form/inputs/inline-edit-input/index.js]: WPCS effect :)

**Pull request merged**